### ISSUE TITLE
Kata voting is not always guaranteed that vote of one account is counted not more than once

### ISSUE THREAD

**Issue by ZEDCWT** [Codewars/codewars.com#1660]
#### Describe the bug
[This kata](https://www.codewars.com/kata/5c25f037cd7f4300150ab3be) is only completed by one user (me) but has 2 'Somewhat Satisfied' votes

#### To Reproduce
So i just finished it at about 12.29 12:10+0800 and tried to gave a 'Somewhat Satisfied' vote
But accidentally clicked on 'Very' just after clicking 'Somewhat' button before it responsed
After it got responsed, i refereshed the page and seeing 
![Pony](https://user-images.githubusercontent.com/7867810/50533838-90c9dd00-0b6d-11e9-923e-6440b6ebb36a.png)
Where the counter shows `75% of 2` instead of `xx% of 1`
And if i switch to 'Somewhat' vote
![Ponx](https://user-images.githubusercontent.com/7867810/50533864-e3a39480-0b6d-11e9-90d7-5304820ec9f6.png)
I thought it was just accidentally count the vote twice so i toggled off the vote and got
![Ponw](https://user-images.githubusercontent.com/7867810/50533895-a68bd200-0b6e-11e9-8ec2-00099041441d.png)
When i re-vote again, it still reports 2 votes

To clearify, im the only one completed that kata and there were no votes before i voted (apparently)
![ponv](https://user-images.githubusercontent.com/7867810/50533901-b99ea200-0b6e-11e9-9f3f-93007c693ef3.png)

In conclusion, a phantom vote is created by such operations.

**Comment by kazk** [comment]
I think I've seen this happen before. Kata voting has some black magic that I don't fully understand yet.

I've fixed the data for this one for now. I'll look into the cause.

The main problem is that the actual votes are stored separately and it _should_ update the value on kata when changed, but sometimes it fails to for various reasons.

**Comment by jhoffner** [comment]
The screenshots shown show 2 completions (the circles icon). Total Code Submissions may only say 1 because the author's doesn't count. 

Authors shouldn't be able to vote on their own kata, thats a bug (or at least a design flaw)

**Comment by kazk** [comment]
The vote wasn't by the author.
The only user with vote was @ZEDCWT and his vote was counted twice. It was "stuck" due to race condition (second vote was added while the first one was still being processed).
I think this is what happened because I was able to fix the data by simply making it recalculate.
